ChatWithAuthSample/AuthorizationSample/Startup.cs

```diff
@@ -0,0 +1,177 @@
+                            Console.WriteLine(accessToken);
+                            // Read the token out of the query string
+                            context.Token = accessToken;
+                        }
+                        return Task.CompletedTask;
+                    },
+                    OnAuthenticationFailed = context =>
+                    {
+                        Console.WriteLine("SigningKey Id {0}", _signingKey.KeyId);
+                        Console.WriteLine("SigningKey {0}", Encoding.ASCII.GetString(_signingKey.Key));
+                        Console.WriteLine("Audience {0}", context.Options.Audience);
+                        Console.WriteLine("Issuer {0}", context.Options.ClaimsIssuer);
+                        return Task.CompletedTask;
+                    },
+                    OnTokenValidated = context =>
+                    {
+                        Console.WriteLine("SigningKey Id {0}", _signingKey.KeyId);
+                        Console.WriteLine("SigningKey {0}", Encoding.ASCII.GetString(_signingKey.Key));
+                        Console.WriteLine("Expected Audience {0}", jwtAppSettingOptions[nameof(JwtIssuerOptions.Audience)]);
+                        Console.WriteLine("Audience {0}", context.Options.Audience);
+                        Console.WriteLine("Expected Issuer {0}", jwtAppSettingOptions[nameof(JwtIssuerOptions.Issuer)]);
+                        Console.WriteLine("Issuer {0}", context.Options.ClaimsIssuer);
+                        return Task.CompletedTask;
+                    }
+                };
+                options.RequireHttpsMetadata = false;
+                options.SaveToken = true;
+                options.TokenValidationParameters = new TokenValidationParameters
+                {
+                    ValidateIssuerSigningKey = true,
+                    ValidIssuer = jwtAppSettingOptions[nameof(JwtIssuerOptions.Issuer)],
+                    ValidateIssuer = false,
+                    IssuerSigningKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(jwtAppSettingOptions["SecretKey"])),
+                    ValidAudience = jwtAppSettingOptions[nameof(JwtIssuerOptions.Audience)],
+                    ValidateAudience = false
+                };
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                app.UseHsts();
+            }
+
+            // Enable Authentication First
+            app.UseAuthentication();
+            // Set CORS Policy, allowing all right now, should be configured for needs.
+            app.UseCors(x => x.AllowAnyOrigin().AllowAnyMethod().AllowAnyHeader().AllowCredentials());
```

Comment:
Have you validated this will work? Later in the .NET Core releases CORS became a bit more restrictive (a good thing), so it'd be good for you to test this out to make sure all the connections are working in both a localhost environment and once published. I think you have to specify at least an origin now.

Reply:
I've tested this both locally and remotely on a VPS I have hosted on 1and1.com. It is definitely not the preferred configuration, however; it alleviates development issues initially. I would always recommend restricting this, but wanted to keep it as simple as possible.

Reply:
Validated in .NET Core 3.0 Preview 6. Had to add localhost to the CORS per your note. Angular App Repo: https://github.com/h5aaimtron/sample_chatWithIdentityClient